environments/openshift/manifests/observatorium-template.yaml

```diff
@@ -909,6 +909,215 @@ objects:
         - name: telemeter-server-tls
           secret:
             secretName: telemeter-server-shared
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    name: prometheus-ams-${NAMESPACE}
+  rules:
+  - nonResourceURLs:
+    - /metrics
+    verbs:
+    - get
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: prometheus-ams-${NAMESPACE}
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: prometheus-ams-${NAMESPACE}
+  subjects:
+  - kind: ServiceAccount
+    name: prometheus-ams
+    namespace: ${NAMESPACE}
+- apiVersion: v1
+  data:
+    nginx.conf: |
+      daemon off;
+      worker_processes 1;
+      error_log /dev/stderr;
+      pid /tmp/nginx.pid;
+
+      events {
+          worker_connections 1024;
+      }
+
+      http {
+        log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                          '$status $body_bytes_sent "$http_referer" '
+                          '"$http_user_agent" "$http_x_forwarded_for"';
+
+        upstream app {
+          server http://thanos-receive.${NAMESPACE}.svc.cluster.local:10902/api/v1/receive;
```

Comment:
There is just one more issue I see here. The pods backing this service can change over time, however NGINX only resolves the DNS name one time and caches it. If the thanos-receive pods get cycled and their IPs change, or if the hashring is re-sized, the cached IPs will be incorrect. We could tell NGINX to limit the cache duration of the DNS resolution to maybe 1 minute?

Reply:
Thanks for pointing this out. After a quick search, I've realized this might be a problem, I'll look into this. https://www.nginx.com/blog/dns-service-discovery-nginx-plus/#domain-name-proxy_pass

Reply:
I have updated the configuration as it suggested in the blog post. I haven't tested it yet.